mysql-test/suite/encryption/r/innodb_encryption_aborted_rotation.result

```diff
@@ -76,8 +76,10 @@ SET GLOBAL debug="+d,crash_on_t1_flush_after_dd_update";
 SET GLOBAL default_table_encryption=ONLINE_FROM_KEYRING_TO_UNENCRYPTED;
 # Make sure that t1 has been decrypted
 # First restart the server after crash so any redo logs for t1 were proceed. If there are any logs available
-# t1 will be opened without validating idb file to DD.
-# restart:--innodb-encryption-rotate-key-age=15 --innodb-encryption-threads=0
+# t1 will be opened without validating idb file to DD. We set here innodb-encryption-threads to value > 0 - so
+# in case there is no redo for t1 and validation to DD would occur we would not get an error in the log
+# to set innodb-encryption-threads to value > 0 to finish rotation. We grep for this error later.
+# restart:--innodb-encryption-rotate-key-age=15 --innodb-encryption-threads=1
```

Comment:
Can you help with the code for below
1. the place where the error message came from "In file './test/t1.ibd' (tablespace id = 4) encryption flag is ON. However the encryption flag in the data dictionary is OFF. This indicates that the rotation of the table was interrupted before space's flags were updated. Please have encryption_thread variable (innodb-encryption-threads) set to value > 0. So the encryption could finish up the rotation."

2. How enabling encryption threads suppress this message?

3. Where is recheck done?

Reply:
2. Enabling encryption threads on restart indicate that they will still have a chance to finish up the encryption rotation. Since the page0 has not yet been updated - encryption threads will take this tablespace and finish up the encryption (updating page0 at the end).
3. Recheck is done on the restart of the server (after the crash) - given encryption threads are on.

Reply:
Ok, I meant the exact location in source code. with some minor effort, it is in Fil_shard::get_file_size().